contracts/mocks/SwapRouterMock.sol

```diff
@@ -0,0 +1,118 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.0;
+
+import {WadRayMath} from "../dependencies/WadRayMath.sol";
+import {SafeCast} from "@openzeppelin/contracts/utils/math/SafeCast.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {ISwapRouter} from "@uniswap/v3-periphery/contracts/interfaces/ISwapRouter.sol";
+import {IERC20Metadata} from "@openzeppelin/contracts/token/ERC20/extensions/IERC20Metadata.sol";
+
+/**
+ * @title SwapRouterMock
+ * @notice SwapRouter mock that can swap a single type of token for several others
+ */
+contract SwapRouterMock is ISwapRouter {
+  using SafeERC20 for IERC20Metadata;
+  using WadRayMath for uint256;
+  using SafeCast for uint256;
+
+  /**
+   * @dev Only one token in is allowed for swaps
+   */
+  IERC20Metadata internal immutable _tokenIn;
+  uint256 internal immutable _wadToTokenInFactor;
+
+  event PriceUpdated(uint256 price);
+
+  uint256 internal _price;
```

Comment:
Y si hacés que el tokenIn y tokenOut puedan ser cualquier cosa, sería más práctico tener en lugar de un solo _price un mapping, donde le seteás el price para determinado in/out. Entonces es más sencillo porque el mock lo podés usar para distintos casos.